app/controllers/api/subcollections/container_images.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Api
+  module Subcollections
+    module ContainerImages
+      def container_images_scan_resource(provider_id, type, image_id, _payload)
+        api_action(type, image_id, :skip_href => true) do |klass|
+          image = resource_search(image_id, type, klass)
+          begin
+            desc = "#{container_image_ident(image)} scanning"
+            task_id = image.scan
+            action_result(true, desc).merge(:task_id => task_id, :parent_id => provider_id)
+          rescue Exception => err
```

Comment:
Use `rescue => e` here, or `rescue SpecificError => e` if there's a specific error you're anticipating (see above for why you might not need to add exception handling here). `Exception` is not a specific error, and attempting to rescue it can cause some unexpected/unpleasant behavior.

Reply:
Codeclimate didn't like `rescue => e` either, So if it stays here it will have to be against its suggestion.

Reply:
I'm OK with that. Handling a specific error is the best practice that Codeclimate is trying to enforce. Rescuing `Exception` is far from a best practice and should be avoided at all cost.

> I don't the conditions that will make this fail but they are probably from outside of the ContainerImage class

If you don't know why you're adding this code, or what the conditions are that will make this fail, or what specific errors you might be anticipating, how can you convince me to merge it? We need answers to these questions (even if you can't encode it in an automated test) as a prerequisite to merging.

Reply:
This is a little similar to [1], But this will go through automate and policy through a series for function calls that will eventually create a scanning Job through `MiqQueue.put` which is the same function that `queue_object_action` eventually calls. In the interest of continuity I suggest we catch errors that same way here, but we can ignore this possibility as well as the code on the `image.scan` function path will only raise an exception if `User.current_user` is not set, which is set for api calls.

[1] https://github.com/ManageIQ/manageiq-api/blob/master/app/controllers/api/vms_controller.rb#L410